### PR TITLE
sublime-music: remove python-Levenshtein

### LINKED ISSUE #212420: sublime-music fails to build: no matching distribution for python-Levenshtein
### Steps To Reproduce
1. build *sublime-music*

### Build log
Just the error:
```
Executing pipInstallPhase
/build/source/dist /build/source
Processing ./sublime_music-0.11.16-py3-none-any.whl
Requirement already satisfied: requests<3.0.0,>=2.24.0 in /nix/store/j865yfw6g27kdsc5jai8w2vl6j46k15i-python3.10-requests-2.28.1/lib/python3.10/site-packages (from sublime-music==0.11.16) (2.28.1)
Requirement already satisfied: peewee<4.0.0,>=3.13.3 in /nix/store/55dgkskz9yqkrlpgvvdmbhpk84qfwgbi-python3.10-peewee-3.15.4/lib/python3.10/site-packages (from sublime-music==0.11.16) (3.15.4)
Requirement already satisfied: PyGObject<4.0.0,>=3.38.0 in /nix/store/xkfrk4sgaqc4c1a1s106apx3r0dkwzm9-python3.10-pygobject-3.42.2/lib/python3.10/site-packages (from sublime-music==0.11.16) (3.42.2)
Requirement already satisfied: python-dateutil<3.0.0,>=2.8.1 in /nix/store/h8qnwa0hzrblmr262b0d1qixi0x63ahi-python3.10-python-dateutil-2.8.2/lib/python3.10/site-packages (from sublime-music==0.11.16) (2.8.2)
ERROR: Could not find a version that satisfies the requirement python-Levenshtein<0.13.0,>=0.12.0 (from sublime-music) (from versions: none)
ERROR: No matching distribution found for python-Levenshtein<0.13.0,>=0.12.0
```
Full log: https://gist.github.com/jshholland/680d93d440d573978ec36d9197aa4778

### Additional context
Built OK as of rev c07552f6f7d4eead7806645ec03f7f1eb71ba6bd. I tried bisecting but too many commits required me to build multiple versions of LLVM/GHC/god knows what else. I also tried applying @dotlambda's patch from https://github.com/sublime-music/sublime-music/pull/370 but that didn't change the error. 

### Notify maintainers

@albakham @sumnerevans 

### Metadata

```console
$ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/rmhwi0jcya8f87gzk2jjdwv4hifmmbb4-nix-info
copying path '/nix/store/rmhwi0jcya8f87gzk2jjdwv4hifmmbb4-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.3, NixOS, 23.05 (Stoat)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - channels(root): `"nixos-21.11pre295280.fa0326ce523"`
 - channels(josh): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by fabaff**
###### Description of changes
Fix build

Fixes #212420

In #208232 `python-Levenshtein` was removed.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 212425` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sublime-music</li>
  </ul>
</details>

**Review by jshholland — APPROVED** [review]
Looks good to me, builds and runs fine on my x86_64-linux box here. Thanks!

**Comment by jshholland** [comment]
There's a typo in the commit message though (`ublime-music`)

**Comment by mweinelt** [comment]
> In #208232 python-Levenshtein was removed.

It was a duplicate package, use `levenshtein` instead.

**Comment by dotlambda** [comment]
> > In #208232 python-Levenshtein was removed.
> 
> It was a duplicate package, use `levenshtein` instead.

It is already using levenshtein. That's why this change is very misleading. Just replace python-Levenshtein with Levenshtein.

**Review by sumnerevans — APPROVED** [review]
I really need to get around to actually making a new release with some additional fixes, but this seems fine for now.

Result of `nixpkgs-review pr 212425` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sublime-music</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 212425` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sublime-music</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Renaming instead of removing now.

**Review by sumnerevans — APPROVED** [review]

File: pkgs/applications/audio/sublime-music/default.nix [hunk 1]
```diff
@@ -38,12 +40,21 @@ python3Packages.buildPythonApplication rec {
     "python-mpv"
   ];
 
+  postPatch = ''
+    sed -i "/--cov/d" setup.cfg
+    sed -i "/--no-cov-on-fail/d" setup.cfg
+    # https://github.com/sublime-music/sublime-music/pull/370
+    # Can be removed in later versions (probably > 0.11.16)
+    substituteInPlace pyproject.toml \
+      --replace 'python-Levenshtein = "^0.12.0"' 'Levenshtein = ">0.12.0"'
```
Comment (wegank) [inline pkgs/applications/audio/sublime-music/default.nix:49]: How about fetchpatching from https://github.com/sublime-music/sublime-music/commit/6c6f7ec72f83d643af8638f9380e6227a01809c5 ?

Comment (wegank) [reply pkgs/applications/audio/sublime-music/default.nix:49]: Also the project has migrated to GitHub.

Comment (sumnerevans) [reply pkgs/applications/audio/sublime-music/default.nix:49]: I migrated from poetry to flit in addition to moving the project to GitHub, so the patch would not work.

I need to figure out what to do about a couple of failing tests and then I'll just make a release and I'll do an overhaul of the module at that point.

**Pull request merged**